jetty-websocket/websocket-core/src/main/java/org/eclipse/jetty/websocket/core/internal/FrameFlusher.java

```diff
@@ -224,6 +230,8 @@ else if (batchBuffer != null && batchSpace >= Generator.MAX_HEADER_LENGTH)
 
         if (flush)
         {
+            for (ByteBuffer bb : buffers)
+                bytesOut.add(bb.limit() - bb.position());
```

Comment:
We now iterate through buffers twice.... perhaps do a manual toArray (below) and calc size and copy to the array in the same loop.